spring-kafka-test/src/test/java/org/springframework/kafka/test/junit/GlobalEmbeddedKafkaTestExecutionListenerTests.java

```diff
@@ -0,0 +1,143 @@
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+import org.junit.platform.launcher.listeners.SummaryGeneratingListener;
+
+import org.springframework.util.DefaultPropertiesPersister;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 3.0
+ */
+public class GlobalEmbeddedKafkaTestExecutionListenerTests {
+
+	@Test
+	void testGlobalEmbeddedKafkaTestExecutionListener() throws IOException {
+		System.setProperty(GlobalEmbeddedKafkaTestExecutionListener.LISTENER_ENABLED_PROPERTY_NAME, "true");
+
+		var brokerProperties = new Properties();
+		brokerProperties.setProperty("auto.create.topics.enable", "false");
+
+		var propertiesFile = File.createTempFile("kafka-broker", ".properties");
+
+		try (var outputStream = new BufferedOutputStream(new FileOutputStream(propertiesFile))) {
+			new DefaultPropertiesPersister().store(brokerProperties, outputStream, "Last entry");
+		}
+
+		System.setProperty(GlobalEmbeddedKafkaTestExecutionListener.BROKER_PROPERTIES_LOCATION_PROPERTY_NAME,
+				"file:" + propertiesFile.getAbsolutePath());
+
+		var discoveryRequest =
+				LauncherDiscoveryRequestBuilder.request()
+						.selectors(DiscoverySelectors.selectClass(TestClass1.class),
+								DiscoverySelectors.selectClass(TestClass2.class))
+						.build();
+
+		var summaryGeneratingListener = new SummaryGeneratingListener();
+		LauncherFactory.create().execute(discoveryRequest, summaryGeneratingListener);
+
+		var summary = summaryGeneratingListener.getSummary();
+
+		try {
+			assertThat(summary.getTestsStartedCount()).isEqualTo(2);
+			assertThat(summary.getTestsSucceededCount()).isEqualTo(1);
+			assertThat(summary.getTestsFailedCount()).isEqualTo(1);
+		}
+		catch (Exception ex) {
+			summary.printFailuresTo(new PrintWriter(System.out));
+			throw ex;
+		}
+
+		System.clearProperty(GlobalEmbeddedKafkaTestExecutionListener.LISTENER_ENABLED_PROPERTY_NAME);
+		System.clearProperty(GlobalEmbeddedKafkaTestExecutionListener.BROKER_PROPERTIES_LOCATION_PROPERTY_NAME);
```

Comment:
Typically, whenever I set system properties in a test and clear them at the end of the test, I wrap the body of the test in a try-finally block and clear the system properties in the `finally` block -- just to be on the safe side in case a particular test fails and the system properties adversely affect other tests in the test suite.